src/renderer/components/cluster-manager/lens-views.ts

```diff
@@ -34,6 +34,17 @@ export async function initView(clusterId: ClusterId) {
   parentElem.appendChild(iframe);
 }
 
+export async function removeView(clusterId: ClusterId) {
```

Comment:
Adding this to the end of `initView` should be enough (without tweaking other files)
```ts
  // auto-clean when cluster removed
  await when(() => !clusterStore.getById(clusterId));
  parentElem.removeChild(iframe)
  lensViews.delete(clusterId)
```

Reply:
Thanks...testing it